controllers/posts_controller.js

```diff
@@ -64,3 +64,36 @@ exports.create = function (req, res, next) {
         res.redirect('/blog')
     }).catch(next)
 };
+
+
+//deletes a record that has a title which matches the slug
+exports.delete = function (req, res, next) {
+
+
+    //find a post with the title equal to the slug and delete it
+    //using req.params.slug instead of given req.params['slug'])[0] as it is cleaner
+    Post.deleteOne({title: req.params.slug}, function(err,result){  //A.deleteOne(conditions, callback)
+        if (err)
+            //if the callback function returns an error, let the next middleware error handler handle it
+            next();
+        //set the status of the operation to 202 to indicate the data was marked for deletion and end the response
+        res.status(202).end();
```

Comment:
you need to encapsulate the `if (err)` because it will hit `next()` and then try to execute `res.status`